### PR TITLE
Track service task(deploy/modify/destroy/rollback/retry/purge) into service_order

### PR DESCRIPTION
**Pull request by baixinsui**
Fixed https://github.com/eclipse-xpanse/xpanse/issues/2031 and https://github.com/eclipse-xpanse/xpanse/issues/2049
 
**Deploy service successfully:**
chrome_e7CeYGNVsf

**Modify service successfully:**
image

**Destroy service successfully:**
image

**Deploy service failed:**
image

**Retry failed deployment failed again:**
image

**Purge failed deployment successfully:**
image

### CODE REVIEW

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Comment by Alice1319** [comment]
LGTM

**Comment by WangLiNaruto** [comment]
LGTM

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 1]
```diff
@@ -397,7 +394,7 @@ private void deployService(DeployTask deployTask) {
                 deployTask.getOcl().getDeployment().getDeployerTool().getKind());
         DeployServiceEntity serviceEntity = storeNewDeployServiceEntity(deployTask);
         ServiceOrderEntity orderTaskEntity =
-                serviceOrderManager.createServiceOrderTask(deployTask, null);
+                serviceOrderManager.createServiceOrderTask(deployTask, serviceEntity);
         try {
             policyValidator.validateDeploymentWithPolicies(deployTask);
             serviceOrderManager.startOrderProgress(deployTask.getOrderId());
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:400]: sure. The new service always be created and stored.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployResult.java [hunk 2]
```diff
@@ -32,12 +32,12 @@ public class DeployResult {
     private UUID serviceId;
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/deployment/DeployResult.java:32]: Deleted.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java [hunk 3]
```diff
@@ -55,29 +55,38 @@ public class ServiceOrderManager {
...
-        orderTask.setDeployServiceEntity(deployServiceStorage
-                .getReferenceById(deployTask.getServiceId()));
+        orderTask.setDeployServiceEntity(deployServiceEntity);
+        orderTask.setOriginalServerId(deployTask.getServiceId());
+        orderTask.setWorkflowId(deployTask.getWorkflowId());
         orderTask.setNewDeployRequest(deployTask.getDeployRequest());
         orderTask.setTaskStatus(TaskStatus.CREATED);
-        if (Objects.nonNull(previousDeployedService)) {
-            orderTask.setPreviousDeployRequest(previousDeployedService.getDeployRequest());
+        if (deployTask.getTaskType() == ServiceOrderType.DEPLOY) {
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceOrderManager.java:74]: deleted

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java [hunk 4]
```diff
@@ -202,27 +208,27 @@ public void updateServiceOrderTaskWithDeployResult(DeployResult deployResult,
...
+                completeParentServiceOrder(storedEntity.getParentOrderId());
+                log.info("Rollback order {} completed, related parent order {} is completed.",
+                        storedEntity.getOrderId(), storedEntity.getParentOrderId());
             }
         }
+    }
+
+    private void completeParentServiceOrder(UUID parentOrderId) {
+        ServiceOrderEntity parentOrder = serviceOrderStorage.getEntityById(parentOrderId);
+        ServiceOrderEntity entityToUpdate = new ServiceOrderEntity();
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java:310]: The previous properties stored into the service_order table when every order entity created, and the new properties after task stored in the table deploy_service. No properties need to update into parent order.

**Comment by WangLiNaruto** [comment]
LGTM

**Pull request merged**